### PR TITLE
nixos/legit: init

### PR DESCRIPTION
**Pull request by ratsclub**
###### Description of changes

Add [legit](https://github.com/icyphox/legit), a web frontend for git, package and module.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/legit-web/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+let
+  inherit (lib)
+    licenses
+    maintainers;
+
+  inherit (maintainers)
+    ratsclub;
+in
+
```
Comment (AndersonTorres) [inline pkgs/applications/version-management/legit-web/default.nix:11]:
```suggestion

```

File: pkgs/applications/version-management/legit-web/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    repo = "legit";
+    owner = "icyphox";
+    rev = "v${version}";
+    sha256 = "sha256-Y0lfbe4xBCj80z07mLFIiX+shvntYAHiW2Uw7h94jrE=";
+  };
+
+  vendorSha256 = "sha256-RAUSYCtP4rcJ2zIBXfPAEZWD1VSfr3d4MrmUMiPpjK8=";
```
Comment (AndersonTorres) [inline pkgs/applications/version-management/legit-web/default.nix:23]:
```suggestion
  vendorHash = "sha256-RAUSYCtP4rcJ2zIBXfPAEZWD1VSfr3d4MrmUMiPpjK8=";
```

File: nixos/tests/legit.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  inherit (maintainers)
+    ratsclub;
+
+  port = 5000;
+  scanPath = "/var/lib/legit";
+in
+{
+  name = "legit-web";
+  meta.maintainers = [ ratsclub ];
```
Comment (AndersonTorres) [inline nixos/tests/legit.nix:14]:
```suggestion
  meta.maintainers = [ lib.maintainers.ratsclub ];
```

File: nixos/tests/legit.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import ./make-test-python.nix ({ lib, pkgs, ... }:
+let
+  inherit (lib)
+    maintainers;
+
+  inherit (maintainers)
+    ratsclub;
+
```
Comment (AndersonTorres) [inline nixos/tests/legit.nix:8]:
```suggestion
```

**Review by AndersonTorres — COMMENTED** [review]
Reorganize the commit history.

File: nixos/modules/services/networking/legit.nix [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+{
+  options.services.${service} = {
+    enable = mkEnableOption (mdDoc "legit git web frontend");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.legit-web;
+      defaultText = literalExpression "pkgs.legit-web";
+      description = mdDoc "legit package to use.";
+    };
```
Comment (thiagokokada) [inline nixos/modules/services/networking/legit.nix:33]:
```suggestion
    package = mkPackageOptionMD pkgs "legit-web" { };
```

**Comment by AndersonTorres** [comment]
Split the commits according to the modules affected.
- one for inclusion of the program itself
- other for the NixOS module
- another for the NixOS tests

**Comment by ratsclub** [comment]
@AndersonTorres is this how you expected the commits to be?

**Review by thiagokokada — APPROVED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Minor fixups; besides, looks good enough.

**Comment by AndersonTorres** [comment]
This should be some legacy code expecting extra options. It deserves further investigation, but not now.

File: nixos/modules/services/networking/legit.nix [hunk 6]
```diff
@@ -0,0 +1,183 @@
+{ config, lib, pkgs }:
```
Comment (AndersonTorres) [inline nixos/modules/services/networking/legit.nix:1]:
```suggestion
{ config, lib, pkgs, ... }:
```

Here the ellipsis are needed.

File: nixos/modules/services/networking/legit.nix [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    mkEnableOption
+    mdDoc
+    mkIf
+    mkOption
+    mkPackageOptionMD
+    optionalAttrs
+    optional
+    types;
+
+  service = "legit";
```
Comment (AndersonTorres) [inline nixos/modules/services/networking/legit.nix:15]:
```suggestion
```

Unneeded.

File: nixos/tests/legit.nix [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import ./make-test-python.nix ({ lib, pkgs }:
```
Comment (AndersonTorres) [inline nixos/tests/legit.nix:1]:
```suggestion
import ./make-test-python.nix ({ lib, pkgs, ... }:
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/legit-web/default.nix [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, ...
```
Comment (AndersonTorres) [inline pkgs/applications/version-management/legit-web/default.nix:4]:
```suggestion
```

Removing this line causes any problem?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Ping me 2 days for merge.

**Pull request merged**